### PR TITLE
Add Versioning Guide to CONTRIBUTING file

### PR DESCRIPTION
**Pull request by rye**
This is a fairly straightforward implementation of the early spec reached in #57, which is still awaiting feedback from @samontea.

Please do provide feedback as you see fit.

### CODE REVIEW

**Comment by cg505** [comment]
I thought we decided to ditch the `release/` branches and just tag instead?

**Comment by rye** [comment]
From [this comment](https://github.com/kotct/dot/issues/57#issuecomment-315226276) it sounded like 

>the `release-` branch can get discarded after the release is released and the branch is in `master`

Basically we'd keep them around for just a short time while bumping versions and potentially adding to changelogs

**Comment by rye** [comment]
But we would still use tags, and the `release-` branches would get deleted after the release took place.

File: CONTRIBUTING.md
Comment (cg505) [inline CONTRIBUTING.md]: typo: "typically"

File: CONTRIBUTING.md
Comment (cg505) [inline CONTRIBUTING.md]: What is the point of the `release-vX.Y` branch? Seems unnecessary to me. The one possible use is leaving one around that always points to the latest release in that minor version, as a convenience. This mostly seems unnecessary though.

File: CONTRIBUTING.md
Comment (cg505) [inline CONTRIBUTING.md]: Imo the PR should target `stable`, and the tag is on that merge commit into `stable`. `stable` then can be merged back into master. It doesn't make sense to have active devel going on in master, make a release branch, and then merge that right back into `master`. The diff should be 0.

**Review by cg505 — CHANGES_REQUESTED** [review]
I disagree with how the release process is put forward. Discuss as needed, but if substantial discussion must be had, move it to the issue instead of having it in the PR.

File: CONTRIBUTING.md
Comment (rye) [reply CONTRIBUTING.md]: Bumping a version is something that should probably not really be done by people submitting changes.  That's project-level work that should really just be done by the maintainers when they know what they want to merge.  We'd delete the branch, but people would be able to create the branch again if we needed to add something to a previous version in a patch release.

Comment (cg505) [reply CONTRIBUTING.md]: So I understand how you would use a `release-v2.1.0` branch, for instance, but what's the point of the `release-v2.1` branch?

Also, can we consider doing `release/` so that we don't have two different types of prefixes?

File: CONTRIBUTING.md
Comment (rye) [reply CONTRIBUTING.md]: I would agree if you're saying that any development work off of `stable` should take place and be directed back to the `stable` branch, but some people like @cg505 and myself would probably be working off of `master`.  Again, making a `release-vX.Y` branch is intended for work _related to the release process_, (incl. merging PR's) and nothing else.  The point of branching off of one of these branches is to have a nice place to merge back into, but I don't expect 3rd parties (or even contributors who are submitting their own changes) to properly take care of documenting their changes.

I don't understand what you mean by
>The diff should be 0.

Comment (cg505) [reply CONTRIBUTING.md]: Popping this back into the issue (#57) for more discussion.

File: CONTRIBUTING.md
Comment (rye) [reply CONTRIBUTING.md]: Further discussion should probably take place on #57.

File: CONTRIBUTING.md
Comment (cg505) [inline CONTRIBUTING.md]: We decided to merge into `stable` via a PR?

File: CONTRIBUTING.md
Comment (cg505) [inline CONTRIBUTING.md]: Add a step to merge the release back into `master`. No PR needed for this one, just do it.

File: CONTRIBUTING.md
Comment (cg505) [reply CONTRIBUTING.md]: Also we should probably get at least one approval (on the release notes and on the release) in order to do this merge. Should already be enforced with our current PR rules. (Don't worry about changing the actual text of the file here, this was just meant to be a note.)

File: CONTRIBUTING.md
Comment (cg505) [inline CONTRIBUTING.md]: >submit a PR for both the `release/vX.Y.Z` and the `hotfix/` branch.

What does it mean to submit a PR for the `hotfix/` branch? Aren't we working in the `hotfix/` branch? We PR `hotfix/*` into `release/vX.Y.Z` and then (if X.Y is the current minor version) PR `release/vX.Y.Z` into stable and tag. Right?

**Review by cg505 — CHANGES_REQUESTED** [review]
Looking pretty good. A few small remaining issues.

File: CONTRIBUTING.md
Comment (rye) [reply CONTRIBUTING.md]: Note that I haven't updated this file at all since our discussion in #57, I was just looking to highlight the actual issues in this file.  Will fix shortly.

File: CONTRIBUTING.md
Comment (rye) [reply CONTRIBUTING.md]: Isn't `master` merge-locked?

File: CONTRIBUTING.md
Comment (rye) [reply CONTRIBUTING.md]: Yeah, that's right.  `hotfix` goes into `release`, `release` goes into `stable`, `stable` gets tagged, `stable` gets merged back to `master`.

File: CONTRIBUTING.md
Comment (rye) [reply CONTRIBUTING.md]: This means we have to have a reviewed PR before we can merge.  That's why I was thinking about doing it the other way; merge to `master` then merge to `stable`.

Comment (cg505) [reply CONTRIBUTING.md]: mmmm. Since an admin will be doing the release, we can just open a PR, then override checks and immediately merge I guess.

File: CONTRIBUTING.md
Comment (cg505) [reply CONTRIBUTING.md]: Okay. Maybe clarify this language to say what is being merged into what in those PRs.

File: CONTRIBUTING.md
Comment (rye) [reply CONTRIBUTING.md]: Yeah, it'd work that way.

File: CONTRIBUTING.md
Comment (rye) [reply CONTRIBUTING.md]: Yep.

**Review by cg505 — APPROVED** [review]
feel free to merge

**Comment by rye** [comment]
Whee! Merging now.

**Pull request merged**
